### PR TITLE
Feature Enhancements and New Pages Integration

### PR DESCRIPTION
**Pull request by r22222**
This PR introduces several updates and feature enhancements to the Job Hunting AI Web Tool project:

	1.	Autocomplete and Chips Selection on the browse all page:
	        •	Implemented advanced filtering with an autocomplete and chips selection feature for better user experience when filtering job results.
	2.	UI/UX Improvements:
	        •	Added padding and improved layout alignment across various components to ensure a consistent and polished user interface.
	        •	Adjusted light and dark mode styles for better readability and aesthetics across all pages.
	3.	New Pages:
	        •	Introduced new “Features,” “About,” and “Sources” pages to provide users with detailed information about the project, its features, and the data sources for the project.
	        •	Integrated images into the “Features” and “About” pages to enhance visual presentation.
	4.	Routing Updates:
	        •	Updated the App component to include routing for the new pages.

### CODE REVIEW

**Comment by bryanw121** [comment]
Feel free to revert my commit, but I mostly just made some stylistic changes that make the table look nicer. I think overall, the PR looks good. 

But some feedback: 
- The sources page doesn't show up in dark mode. 
- The filtering by salary doesn't really make sense right now, because it's not a numeric filter. 
- Feel free to refactor the styling changes I made, I just kinda made them on a whim bc I still want things to look decent aesthetically. 

Good work overall!

**Comment by r22222** [comment]
I think i resolved the issues...ran into some issues with a rebase and still no 100% what happened so let me know if something broke...Hoping we can get this filter approved and move on!

File: frontend/src/Pages/Home/InputForm/InputForm.tsx [hunk 1]
```diff
@@ -13,8 +13,10 @@ import {
   Select,
```
Comment (bryanw121) [inline frontend/src/Pages/Home/InputForm/InputForm.tsx:13]: looks like the input form in this PR is using an old version. Please check the files and what you're changing in your PR.

**Review by bryanw121 — CHANGES_REQUESTED** [review]
changes are okay, but there are a lot of merge conflicts and incorrect versions

File: frontend/src/Pages/Home/InputForm/InputForm.tsx (hunk 1 above)
Comment (r22222) [reply frontend/src/Pages/Home/InputForm/InputForm.tsx]: updated

File: backend/app.py [hunk 2]
```diff
@@ -127,4 +130,5 @@ def get_recommendation():
         err = f"ERROR: {e}"
         sys.exit()
 
+    app.run(host='0.0.0.0', port=8080)
```
[the remark this answers is not in the document]
Comment (r22222) [reply backend/app.py]: updated

File: backend/app.py [hunk 3]
```diff
@@ -1,7 +1,10 @@
 import sys
 import os
 sys.path.append(os.path.dirname(os.path.abspath(__file__)))
+from flask import Flask, make_response, jsonify
+=======
 from flask import Flask, make_response, jsonify, request
+>>>>>>> origin
```
[the remark this answers is not in the document]
Comment (r22222) [reply backend/app.py]: fixed.

File: .gitignore [hunk 4]
```diff
@@ -1 +1,2 @@
-venv/
\ No newline at end of file
+venv/
+venv/
```
[the remark this answers is not in the document]
Comment (r22222) [reply .gitignore:2]: fixed.

File: frontend/app.yaml [hunk 5]
```diff
@@ -1,4 +1,5 @@
 runtime: nodejs18
+service: frontend-service
```
[the remark this answers is not in the document]
Comment (r22222) [reply frontend/app.yaml:2]: I'm not sure what this is doing here...

File: frontend/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,18 @@
+FROM node:20.9.0-bullseye
```
[the remark this answers is not in the document]
Comment (r22222) [reply frontend/Dockerfile:1]: IDK about this one either...

**Review by bryanw121 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (r22222) [reply .gitignore]: I believe this is actually done now

File: frontend/Dockerfile (hunk 6 above)
[the remark this answers is not in the document]
Comment (r22222) [reply frontend/Dockerfile]: removed!

File: frontend/app.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (r22222) [reply frontend/app.yaml]: removed!

**Review by bryanw121 — APPROVED** [review]

**Pull request merged**